test/jest/victory-legend/victory-legend.test.js

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable max-nested-callbacks */
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { VictoryLegend } from "victory-legend";
+import { isCircle, isTriangle } from "../../svg-test-helper";
+
+describe("components/victory-legend", () => {
+  const initialData = [
+    {
+      name: "Series 1",
+      symbol: {
+        type: "circle"
+      }
+    },
+    {
+      name: "Series 2",
+      labels: {
+        fill: "red"
+      },
+      symbol: {
+        type: "triangleUp",
+        fill: "blue"
+      }
+    }
+  ];
+
+  it("renders provided data correctly", () => {
+    const { container } = render(<VictoryLegend data={initialData} />);
+    const legendLabels = container.querySelectorAll("text");
+
+    expect(legendLabels).toHaveLength(initialData.length);
+  });
+
+  it("has expected horizontal label position", () => {
+    const { container } = render(
+      <VictoryLegend data={initialData} orientation="horizontal" />
+    );
+
+    const [label1, label2] = container.querySelectorAll("text");
+
+    expect(label1.getAttribute("y")).toEqual(label2.getAttribute("y"));
+  });
+
+  it("has expected vertical symbol position", () => {
+    const { container } = render(
+      <VictoryLegend data={initialData} orientation="vertical" />
+    );
+
+    const [label1, label2] = container.querySelectorAll("text");
+
+    expect(label1.getAttribute("x")).toEqual(label2.getAttribute("x"));
+  });
+
+  describe("symbols", () => {
+    const legendData = [
+      {
+        name: "Series 1",
+        labels: {
+          fontSize: 10
+        },
+        symbol: {
+          type: "circle",
+          fill: "red"
+        }
+      },
+      {
+        name: "Long Series Name",
+        labels: {
+          fontSize: 12
+        },
+        symbol: {
+          type: "triangleUp",
+          fill: "blue"
+        }
+      }
+    ];
+
+    it("has expected symbols length", () => {
+      const { container } = render(<VictoryLegend data={legendData} />);
+      const symbols = container.querySelectorAll("path");
+      expect(symbols).toHaveLength(2);
+    });
+
+    it("has expected symbol colors", () => {
+      const { container } = render(<VictoryLegend data={legendData} />);
+      const symbols = container.querySelectorAll("path");
+
+      symbols.forEach((symbol, index) => {
+        const style = symbol.getAttribute("style");
+        expect(style).toContain(legendData[index].symbol.fill);
+      });
+    });
+
+    it("has expected symbol type", () => {
+      const { container } = render(<VictoryLegend data={legendData} />);
+      const [circleSymbol, triangleSymbol] = Array.from(
+        container.querySelectorAll("path")
+      ).map((symbol) => symbol.getAttribute("d"));
+
+      expect(isCircle(circleSymbol)).toBeTruthy();
+      expect(isTriangle(triangleSymbol)).toBeTruthy();
```

Comment:
This is probably our best bet for now. In the future, I'm not even sure how useful this kind of test is since we also have chromatic to ensure that these paths are not changing shape.